### PR TITLE
WIP: implement {array_, VHD}split

### PR DESCRIPTION
**Pull request by ev-br**
The main difference with pytorch is that numpy expects `indices_or_sections` as indices into the array to split, while pytorch takes lengths of sections. Also some differences in what exactly happens when the length of the array to split is not equally divisible into sections. And some edge cases for when there are empty sections.

### CODE REVIEW

**Comment by ev-br** [comment]
Also `tile` and `kron`, let's pick low-hanging fruit now that I'm looking at this test file.

File: torch_np/_detail/implementations.py [hunk 1]
```diff
@@ -15,3 +15,63 @@ def tensor_equal(a1_t, a2_t, equal_nan=False):
...
+
+def split_helper_list(tensor, indices_or_sections, axis, strict=False):
+    if not isinstance(indices_or_sections, list):
+        raise NotImplementedError("split: indices_or_sections: list")
+    # numpy expectes indices, while torch expects lengths of sections
+    # also, numpy appends zero-size arrays for indices above the shape[axis]
+    lst = [x for x in indices_or_sections if x <= tensor.shape[axis]]
+    num_extra = len(indices_or_sections) - len(lst)
+
+    lst = lst + [tensor.shape[axis]]
```
Comment (lezcano) [inline torch_np/_detail/implementations.py:71]: Nit. prefer append to avoid creating a new list.

File: torch_np/_wrapper.py [hunk 2]
```diff
@@ -158,6 +159,77 @@ def stack(arrays, axis=0, out=None, *, dtype=None, casting="same_kind"):
...
 
 
+def array_split(ary, indices_or_sections, axis=0):
+    tensor = asarray(ary).get()
+    base = ary if isinstance(ary, ndarray) else None
+    axis = _util.normalize_axis_index(axis, tensor.ndim)
+
+    result = _impl.split_helper(tensor, indices_or_sections, axis)
+
+    return tuple(maybe_set_base(_, base) for _ in result)
```
Comment (lezcano) [inline torch_np/_wrapper.py:169]: `_` is used for variables that are not used, but here it's being used. Rename it to `x`

File: torch_np/_wrapper.py [hunk 3]
```diff
@@ -158,6 +159,77 @@ def stack(arrays, axis=0, out=None, *, dtype=None, casting="same_kind"):
...
+
+
+def split(ary, indices_or_sections, axis=0):
+    tensor = asarray(ary).get()
+    base = ary if isinstance(ary, ndarray) else None
```
Comment (lezcano) [inline torch_np/_wrapper.py:174]: In general, I think that `base` should be dropped. Too big a hassle for an almost-negligible win. It also makes the wrapper heavier, which is not good.

**Review by lezcano — APPROVED** [review]
Cool, just a few minor points.

File: torch_np/_wrapper.py (hunk 3 above)
Comment (ev-br) [reply torch_np/_wrapper.py:174]: Let me open https://github.com/Quansight-Labs/numpy_pytorch_interop/issues/47 and push the decision to a latter stage.

**Comment by ev-br** [comment]
Thanks Mario for the review!

File: torch_np/_wrapper.py (hunk 3 above)
Comment (lezcano) [reply torch_np/_wrapper.py:174]: for sure
